pcdsdevices/gasdet.py

```diff
@@ -0,0 +1,345 @@
+"""
+Ophyd devices for the acqiris digitizer systems
+"""
+import logging
+from ophyd import EpicsSignal, EpicsSignalRO, Device
+from ophyd import Component as Cpt
+from ophyd import FormattedComponent as FCpt
+from .interface import BaseInterface
+
+logger = logging.getLogger(__name__)
+
+
+class acqiris_channel(Device):
+    """
+    Class to define a single acqiris channel.
+    """
+    waveform = Cpt(
+        EpicsSignalRO,
+        ':Data',
+        kind='hinted'
+    )
+    v_max = Cpt(
+        EpicsSignal,
+        ':CMaxVert',
+        kind='normal'
+    )
+    v_cent = Cpt(
+        EpicsSignal,
+        ':CCenter',
+        kind='normal'
+    )
+    v_in = Cpt(
+        EpicsSignal,
+        ':CMinVert',
+        kind='normal'
+    )
+    v_scale = Cpt(
+        EpicsSignal,
+        ':CVPerDiv',
+        kind='normal'
+    )
+    bandwidth = Cpt(
+        EpicsSignal,
+        ':CBandwidth',
+        kind='config',
+        string=True
+    )
+    full_scale = Cpt(
+        EpicsSignal,
+        ':CFullScale',
+        kind='config'
+    )
+    coupling = Cpt(
+        EpicsSignal,
+        ':CCoupling',
+        kind='config',
+        string=True
+    )
+    offset = Cpt(
+        EpicsSignal,
+        ':COffset',
+        kind='config'
+    )
+    trig_coupling = Cpt(
+        EpicsSignal,
+        'CTrigCoupling',
+        kind='config',
+        string=True
+    )
+    trig_slope = Cpt(
+        EpicsSignal,
+        'CTrigSlope',
+        kind='config',
+        string=True
+    )
+    trig_level = Cpt(
+        EpicsSignal,
+        ':CTrigLevel1',
+        kind='config'
+    )
```

Comment:
The class is extremely hard to read when you take newlining to the 11th level like this, if only because every other device class's components are laid out on 1 line (or 2 if unfortunate)

Reply:
PEP8 has a 79 character limit so that code looks the same for people on any terminal.  OK. Fine.  It can look terrible for everybody.

Reply:
You can adhere to pep8 while not making your file have lines of length 25

Reply:
Interestingly, `black` (an automatic code formatter) which is generally _awful_ with putting everything on separate lines and vertical spacing, has put your components on single lines:

```python
class AcqirisChannel(Device):
    """
    Class to define a single acqiris channel.
    """

    waveform = Cpt(EpicsSignalRO, ":Data", kind="hinted")
    v_max = Cpt(EpicsSignal, ":CMaxVert", kind="normal")
    v_cent = Cpt(EpicsSignal, ":CCenter", kind="normal")
    v_in = Cpt(EpicsSignal, ":CMinVert", kind="normal")
    v_scale = Cpt(EpicsSignal, ":CVPerDiv", kind="normal")
    bandwidth = Cpt(EpicsSignal, ":CBandwidth", kind="config", string=True)
    full_scale = Cpt(EpicsSignal, ":CFullScale", kind="config")
    coupling = Cpt(EpicsSignal, ":CCoupling", kind="config", string=True)
    offset = Cpt(EpicsSignal, ":COffset", kind="config")
    trig_coupling = Cpt(EpicsSignal, "CTrigCoupling", kind="config", string=True)
    trig_slope = Cpt(EpicsSignal, "CTrigSlope", kind="config", string=True)
    trig_level = Cpt(EpicsSignal, ":CTrigLevel1", kind="config")
```

You might consider running it (`pip install black; black pcdsdevices/gasdet.py`) before going forward. Or not 🤷‍♂

Reply:
Thank you @klauer.  Unfortunately I still have 14 E501 flake8 errors after evoking `black` :(

Reply:
Hopefully the edits with the help of `black` (thanks @klauer) and some tinkering to fix the line lengths is satisfactory?  We have very serious LCLS deadlines and I don't wish to spend all morning tab-indenting code that is probably going to be changed anyway.

Reply:
If you don't have time to get this merged properly, it is ok to run out of dev... I will insist on doing this right the first time though to merge into the main repo.

Reply:
These lines should all look something like:
`trig_coupling = Cpt(EpicsSignal, "CTrigCoupling", kind="config", string=True)`

or, if it rolls above 79:
```
trig_coupling = Cpt(EpicsSignal, "CTrigCoupling",
                    kind="config", string=True)
```